web/src/components/netflow-topology/__tests__/element-panel.spec.tsx

```diff
@@ -0,0 +1,39 @@
+import { DrawerCloseButton } from '@patternfly/react-core';
+import { BaseNode } from '@patternfly/react-topology';
+import { shallow } from 'enzyme';
+import * as React from 'react';
+import { TopologyMetrics } from '../../../api/loki';
+import { MetricFunction, MetricType } from '../../../model/flow-query';
+import { ElementPanel } from '../element-panel';
+import { response } from '../__tests-data__/metrics';
+
+describe('<ElementPanel />', () => {
+  const mocks = {
+    element: {
+      getData: () => ({
+        name: 'loki-distributor-loki-76598c8449-csmh2',
+        addr: '10.129.0.15'
+      }),
+      getType: () => 'Node'
+    } as BaseNode,
+    metrics: response.data.result as TopologyMetrics[],
+    metricFunction: 'sum' as MetricFunction,
+    metricType: 'bytes' as MetricType,
+    onClose: jest.fn(),
+    id: 'element-panel-test'
+  };
+
+  it('should render component', async () => {
+    const wrapper = shallow(<ElementPanel {...mocks} />);
+    expect(wrapper.find(ElementPanel)).toBeTruthy();
+    expect(wrapper.find('#element-panel-test')).toHaveLength(1);
+  });
+
+  it('should close on click', async () => {
+    const wrapper = shallow(<ElementPanel {...mocks} />);
+    const closeButton = wrapper.find(DrawerCloseButton);
+    expect(closeButton).toHaveLength(1);
+    closeButton.simulate('click');
+    expect(mocks.onClose).toHaveBeenCalled();
+  });
```

Comment:
I think that, in order to minimize regressions, maybe we should do other tests, e.g. populating the mock with 2-3 topology metrics and verify that they are rendered properly, including all the conditional renders I see below.

Reply:
sure, I have added basic testing on addresses, from / to / total counters for both `node` & `edge` case